form/src/main/java/com/ritense/form/domain/FormIoFormDefinition.java

```diff
@@ -320,42 +327,50 @@ private boolean isDocumentContentVar(JsonNode field) {
-            if (key.startsWith(externalFormFieldType.name().toUpperCase())) {
-                return false;
-            }
+        if (getExternalFormFieldType(field).isPresent()) {
+            return false;
         }
-
         return !key.isEmpty() && !key.startsWith(PROCESS_VAR_PREFIX.toUpperCase());
     }
 
     private Optional<ExternalContentItem> getExternalFormField(JsonNode field) {
         return getExternalFormFieldType(field).flatMap(externalFormFieldType -> {
             String jsonPath = field.get(PROPERTY_KEY).asText().replace(".", "/");
-            String name = jsonPath.substring(externalFormFieldType.name().length() + 1);//example pv.varName -> gets varName
+            // Note: Name is limited to 1 level of depth. Not support = pv.aa.bbb
+            String name = jsonPath.substring(externalFormFieldType.length() + 1); // example pv.varName -> gets varName
             jsonPath = "/" + jsonPath;
-            return buildJsonPointer(jsonPath).flatMap(jsonPointer -> Optional.of(new ExternalContentItem(name, jsonPointer, externalFormFieldType)));
+            return buildJsonPointer(jsonPath)
+                .flatMap(jsonPointer -> Optional.of(
+                        new ExternalContentItem(
+                            name,
+                            jsonPointer,
+                            externalFormFieldType
+                        )
+                    )
+                );
         });
     }
 
     private boolean isExternalFormField(JsonNode field) {
+        return getExternalFormFieldType(field).isPresent();
+    }
+
+    private Optional<String> getExternalFormFieldType(JsonNode field) {
         if (!field.has(PROPERTY_KEY) && !field.get(PROPERTY_KEY).asText().isEmpty()) {
-            return false;
+            return Optional.empty();
         }
-        var propertyValue = field.get(PROPERTY_KEY).asText().toUpperCase();
-        for (ExternalFormFieldType externalFormFieldType : ExternalFormFieldType.values()) {
-            if (propertyValue.startsWith(externalFormFieldType.name().toUpperCase())) {
-                return true;
-            }
+        final String key = field.get(PROPERTY_KEY).asText().toUpperCase();
+        // Note key can be -> "ExternalFormFieldTypeName.propertyName"
+        if (!key.contains(EXTERNAL_FORM_FIELD_TYPE_SEPARATOR)) {
+            return Optional.empty();
         }
-        return false;
-    }
-
-    private Optional<ExternalFormFieldType> getExternalFormFieldType(JsonNode field) {
-        var propertyValue = field.get(PROPERTY_KEY).asText().toUpperCase();
-        for (ExternalFormFieldType externalFormFieldType : ExternalFormFieldType.values()) {
-            if (propertyValue.startsWith(externalFormFieldType.name().toUpperCase())) {
-                return Optional.of(externalFormFieldType);
+        // Check if key prefix is supported by a resolver.
+        final var resolvers = SpringContextHelper.getBeansOfType(FormFieldDataResolver.class);
```

Comment:
Generally speaking this is a security risk, as now there's a static method that can retrieve any bean at every point in the code. I suggest making a package private variation of this `SpringContextHelper` that only retrieves `FormFieldDataResolver` beans.

Reply:
Will do.